### PR TITLE
generic-types-go: Added DockerImage.Validate

### PR DESCRIPTION
**Pull request by ewoutp**
See subject

### CODE REVIEW

**Comment by ewoutp** [comment]
@zyndiecate RFR

File: image.go [hunk 1]
```diff
@@ -64,6 +64,13 @@ func (img DockerImage) DefaultLatestVersion() DockerImage {
 	return img
 }
 
+// Validate checks that the given docker image is valid.
+// Returns nil if valid, or an error if not valid.
+func (img DockerImage) Validate() error {
```
Comment (zyndiecate) [inline image.go]: How is validation done before? Or why do we need that?

Comment (ewoutp) [inline image.go]: This is needed to validate a docker image after a field of the struct has been changed.
See giantswarm/app-service#306

Comment (zyndiecate) [inline image.go]: Ok cool, but where is this used now?

Comment (ewoutp) [inline image.go]: https://github.com/giantswarm/app-service/pull/315/files#diff-d8b38722a5f501b8c888f9363fed7004R465

Comment (zyndiecate) [inline image.go]: I see, just wondered, because I was not pinged there. All good.

**Comment by zyndiecate** [comment]
LGTM

**Pull request merged**